sensor/common/compliance/service_impl.go

```diff
@@ -232,7 +232,7 @@ func (s *serviceImpl) Communicate(server sensor.ComplianceService_CommunicateSer
 			log.Errorf("receiving from compliance %q: %v", hostname, err)
 			// Make sure the stopper stops if there is an error with the connection
 			stopper.Client().Stop()
-			return err
+			return errors.Wrapf(err, "receiving from compliance %q", hostname)
```

Comment:
This one is good! We add context, but not repeat the fact about this being an error or a failed operation.